azure-pipelines.yml

```diff
@@ -221,13 +225,16 @@ stages:
         kubectl cp $(NAMESPACE)/$(POD_NAME):/mnt/results.xml k8s-results.xml
       env:
         KUBECONFIG: $(KUBECONFIGFILE)
+      displayName: 'Copy the results from the test job pod.'
+    # Finally, we can publish the test results.
     - task: PublishTestResults@2
       condition: succeeded()
       inputs:
         testResultsFormat: 'JUnit'
         testResultsFiles: 'k8s-results.xml'
         testRunTitle: 'Publish kubernetes test results'
         failTaskOnFailedTests: true 
+    # And do a lot of cleaning.
```

Comment:
All the cleaning could be done in one step if everything shared a label?

Reply:
I'll clean all the test resources (the PVC, job and pod) using a label they have in common, but will keep separate the deployment cleaning in a separate step.
They currently do not have a label in common.